engine_comparison/dispatcher.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+# Copyright 2017 Google Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+
+. $(dirname $0)/../common.sh
+
+build_engine() {
+  # [[ ! -e ~/$FENGINE_CONFIGS_DIR/$1 ]] && echo "cant do" && exit 1
+
+  # All of this code will probably be in a separate script
+
+  FENGINE_CONFIG=$1
+  . $FENGINE_CONFIG
+  echo "Building $FENGINE_CONFIG"
+  # export ENGINE_DIR=${ENGINE_DIR}
+}
+
+build_benchmark_using() {
+  BENCHMARK=$1
+  FENGINE_CONFIG=$2
+
+  # Write a new script here, probably
+  #
+  . $FENGINE_CONFIG
+
+  BUILDING_DIR=RUN-${THIS_BENCHMARK}
```

Comment:
Looks like `${THIS_BENCHMARK}` is defined in the caller. Why is it passed as an environment variable while `BENCHMARK` and `FENGINE_CONFIG` are passed as arguments?